README.md
### FIFA 22 Career Mode Cheat Table
**Don't use it in online game modes!**

***
[What is Cheat Table?](https://en.wikipedia.org/wiki/Cheat_Engine#Cheat_Tables)

```
"Cheat Tables" is a file format used by Cheat Engine to store data such as cheat addresses, scripts including Lua scripts and code locations
```

I recommend to use [Cheat Engine 7.3](https://cheatengine.org/downloads.php) or newer for the FIFA 22 cheat table.

***
![](https://i.imgur.com/A1CvRRl.png)

The FIFA 22 Career Mode Cheat Table allows you to modify many aspects of you career mode.
For example you can edit players (potential, attributes, name etc.), transfer budget, boost/nerf players development, play single career for more than 15 seasons, change weather/stadium/time of the day before match, send youth academy scout to any country... and many more!

***

![](https://i.imgur.com/s7KsyrO.png)

This cheat table has also an graphical user interface (GUI) which makes editing a little bit easier for inexperienced users.

***

![](https://i.imgur.com/acsVuVm.png)
![](https://i.imgur.com/93pI9wG.png)
![](https://i.imgur.com/j6EivOU.png)
![](https://i.imgur.com/YqDq4xa.png)
![](https://i.imgur.com/EAwprpX.png)
![](https://i.imgur.com/go6CoXt.png)

![](https://i.imgur.com/bRhxh4u.png)
![](https://i.imgur.com/zNKDnFY.png)
![](https://i.imgur.com/HZ0Jvpa.png)
![](https://i.imgur.com/vTpHpAv.png)

![](https://i.imgur.com/ew3qoa3.png)
![](https://i.imgur.com/zDA9VCX.png)

***
### Useful links
* [Wiki](https://github.com/xAranaktu/FIFA-22-CM-Cheat-Table/wiki)
* [FIFA Projects Board](https://app.gitkraken.com/glo/board/YT9S8HnT7gBLX53M)
* [Release Schedule](https://docs.google.com/spreadsheets/d/1pamwGpr0PNwZjDNMYK2NZmtwIL7lVkQqJAMo702BtNk/edit#gid=0)
* [FIFA 22 IDs](https://docs.google.com/spreadsheets/d/12A4UR9nBNTZw4Z8mW54pSkl4NpVvSOr04v4FABk0-DE/edit#gid=0)
* [Discord server](https://discord.gg/va9EtdB)


changelog.txt
v22.1.1.6:
    - Fixed:
        - "Play As Player -> Manager Reputation"
        - "Play As Player -> Squad Role"
        - "Play As Player -> Create Transfer Offer"
    - Added lua scripts to "users scripts" folder:
        - "extend_user_team_players_contracts.lua"
    - Removed outdated scripts.

v22.1.1.5:
    - Added scripts to "users scripts" folder:
        - "export_transfer_history.lua"

v22.1.1.4:
    - Added to "Play As Player -> Create Transfer Offer":
        - Date
        - IsLoan
    - Fixed "Play As Player -> Manager Reputation"
    - Fixed "Play As Player -> Squad Role"
    - Replaced "Play As Player -> Create Transfer Offer" script with pointers.

v22.1.1.3:
    - Added TU3, 3 new headmodels to "fix_players_headmodels.lua"
    - Fixed "Play As Player->Create Transfer Offer"

v22.1.1.2:
    - Added to "Training" -> "Manager CM":
        - "Always A"
    - Added scripts to "users scripts" folder:
        - "export_season_stats.lua"

v22.1.1.1:
    - Added FIFA 22 Bodytypecodes
    - Added to "Play As Player":
        - Manager Reputation
        - Squad Role

v22.1.1.0:
    - Fixed error when cloning new FIFA 22 icons (Beckham, Cafu, Rooney, Van Persie, Casillas)

v22.1.0.9:
    - Fixed "Editing Player" -> "Player Data" script
    - Added to GUI:
        - "Transfer Players"
        - "Match Schedule Editor"
    - Added scripts to "users scripts" folder:
        - fix_players_headmodels.lua

v22.1.0.8:
    - Added "Clone" tab to "Players Editor"
    - Added to "Non-CM Related":
        - "Game Settings" -> "Camera"
        - "Free Fly Camera"

v22.1.0.7:
    - Update shoe list in "randomize_shoe_models.lua" script
    - Added "Teams Editor":
        - Team Info
        - Formation (Doesn't work in Player Career)
        - Manager
        - Kits
    - Added "Player Contract":
        - Contract Negotiation
        - Disable negotiation status checks

v22.1.0.6:
    - Fixed "Youth Academy" -> Age Range
    - Added to "Youth Academy":
        - Generate new report
        - Generate players with custom ID
        - Potential Range
        - 100% chance for 5* weak foot
        - 100% chance for 5* skill moves
        - Attribute Modifier Range

v22.1.0.5:
    - Added "Training":
        - "Created Player CM" -> "Grade Achieved XP"
    - Moved "Players Development" to "Training" -> "Manager CM"
    - Added to "Play As Player":
        - 999 Skill Points
    - Fixed assigning release clause to player that don't have one.
    - Changed "15 Players Per Report" (now it's midfunc hook)
    - Changed "Reveal Ovr & Pot" (now it's midfunc hook)
    - Added to "Sharpness"
        - "Set at 100 on change"


v22.1.0.4:
    - Added "Play As Player"
        - Positive decision on transfer request
        - Create Transfer Offer
    - Added to "Youth Academy"
        - Settings
            - Age Range Low
            - Age Range High
            - Min Age For Promotion
        - 15 Players Per Report
        - Free Missions
        - Revel Ovr & Pot
    - Added "Sim Match Settings"
        - Max Cards
        - Perc Chance
        - Perc Drop
        - Chance Of Red
        - Chance Of Two Reds
        - Max Injures
        - Don't lose Stamina
        - Base Fatigue
    - Added "Match Settings"
        - Change Stadium
        - Half Length
        - Time Of Day
        - Weather
    - Added "Fitness"
        - "Set at 100 on change"
        - "Heal All Players in your team"
        - "Refill all players energy in your team"
    - Added "Editing Player"
        - Player Data
            - Name
            - Birthdate
            - Kit Number

v22.1.0.3:
    - Added to "Players Editor"
        - Injury Status
        - Fitness
        - Release Clause
        - Sharpness
        - Morale
        - Form
    - Added "Morale"
        - Mass edit players morale in team
    - Added "Form"
        - Mass edit players form in team
    - Added "Sharpness"
        - Mass edit players sharpness in team
        - Mass edit players sharpness in game
    - Added "Job Offers"
        - Club Team
        - National Team
    - Added "Manager"
        - Manager Never Retire
        - Manager Rating - 99
        - "Make ManagerID editable"
    - Added scripts to "users scripts" folder:
        - "1ovr_99pot.lua"
        - "99ovr_99pot.lua"
        - "99ovr_99pot_in_given_team.lua"
        - "99pot_in_given_team.lua"
        - "99pot.lua"
        - "custom_headasset_to_manager.lua"
        - "custom_headasset_to_playerids.lua"
        - "custom_tattoos_to_playerids.lua"
        - "isretiring=0.lua"
        - "isretiring=1.lua"
        - "medium_socklenghtcode.lua"
        - "modifier=0.lua"
        - "randomize.lua"
        - "randomize_shoe_models.lua"
        - "mass_edit_age.lua"
        - "set_generic_heads.lua"
        - "set_generic_heads_alt.lua"
        - "untuck_shirts.lua"

v22.1.0.2:
    - Fixed default start date (changed year from 2020 to 2021)
    - Added "Transfer HUB"
        - Change Release Clause value to 1
        - Allow Transfer Approach
        - Allow Loan approach
    - Added to "Gameplay"
        - Unlimited Substitutions (USER)
        - Home Team
            - Disable Substitutions (CPU)
        - Away Team
            - Disable Substitutions (CPU)

v22.1.0.1:
    - Added "Players Development"
    - Added "Side Changer" (CPU vs CPU)
    - Added "Club Finances"
        - Transfer Budget
    - Added "Youth Academy"
        - Send scout to any country
        - Allow multiple scouts in same country
    - Added "Hire Scouts"
        - "Free 5/5 Scouts"
        - Scout Detailed Info
    - Added "Free & Unlimited Releasing Players"
    - Added "Global Transfer Network"
        - Reveal player data (scouting players not needed)
    - Added "Non-CM Related":
        - "Dont Pause Game on AltTab
    - Added to "Gameplay"
        - Match Timer
        - Never Tired Player IDs
        - Home Team
            - Injured players
            - 5 stars skill moves

users scripts/randomize.lua
--- HOW TO USE:
--- https://i.imgur.com/xZMqzTc.gifv
--- 1. Open Cheat table as usuall and enter your career.
--- 2. In Cheat Engine click on "Memory View" button.
--- 3. Press "CTRL + L" to open lua engine
--- 4. Then press "CTRL + O" and open this script
--- 5. Click on 'Execute' button to execute script and wait for 'done' message box.

--- AUTHOR: ARANAKTU

--- Fun script, don't recommend to use it in long term careers.
--- High chance that it will broke your squad file/career save

--- This script will randomize position, attributes, ovr, potential, age, nationality of all players.


local random_position = true
local random_attributes = true
local random_age = true
local random_nationality = true

local attribute_min = 21
local attribute_max = 99

local age_change_range_min = -5
local age_change_range_max = 5
local age_max = 40  -- Not older than 40 years
local age_min = 16  -- Not younger than 16 years


-- Don't change anything below
local playersEditorFormManager = require 'lua/GUI/forms/playerseditorform/manager';

gCTManager:init_ptrs()
local game_db_manager = gCTManager.game_db_manager
local memory_manager = gCTManager.memory_manager

local nationalities = {
    149,
    1,
    97,
    194,
    2,
    98,
    62,
    63,
    52,
    3,
    64,
    195,
    4,
    5,
    65,
    150,
    151,
    66,
    6,
    7,
    67,
    99,
    68,
    152,
    53,
    8,
    100,
    54,
    69,
    153,
    9,
    101,
    102,
    225,
    154,
    103,
    70,
    104,
    71,
    105,
    106,
    55,
    155,
    213,
    56,
    214,
    107,
    110,
    196,
    72,
    10,
    73,
    85,
    11,
    12,
    108,
    13,
    109,
    74,
    207,
    57,
    111,
    76,
    14,
    112,
    113,
    208,
    142,
    114,
    16,
    197,
    17,
    18,
    115,
    116,
    20,
    21,
    117,
    205,
    22,
    206,
    77,
    157,
    78,
    118,
    119,
    79,
    80,
    34,
    81,
    158,
    23,
    24,
    159,
    160,
    161,
    162,
    26,
    27,
    82,
    163,
    164,
    165,
    120,
    166,
    167,
    219,
    168,
    169,
    170,
    28,
    171,
    121,
    122,
    123,
    29,
    30,
    31,
    172,
    124,
    125,
    173,
    174,
    126,
    32,
    127,
    128,
    83,
    33,
    175,
    15,
    84,
    129,
    130,
    176,
    131,
    177,
    215,
    198,
    86,
    132,
    133,
    19,
    35,
    36,
    178,
    179,
    180,
    87,
    199,
    58,
    59,
    181,
    37,
    38,
    88,
    182,
    25,
    39,
    40,
    134,
    200,
    41,
    183,
    42,
    136,
    51,
    137,
    138,
    184,
    43,
    44,
    201,
    139,
    140,
    218,
    45,
    185,
    89,
    90,
    91,
    141,
    92,
    46,
    47,
    186,
    135,
    202,
    187,
    143,
    188,
    212,
    144,
    203,
    93,
    145,
    48,
    189,
    94,
    96,
    146,
    49,
    190,
    95,
    60,
    191,
    204,
    61,
    192,
    50,
    193,
    147,
    148
}

local attributes_fields = {
    "gkdiving",
    "gkhandling",
    "gkkicking",
    "gkpositioning",
    "gkreflexes",
    "crossing",
    "finishing",
    "headingaccuracy",
    "shortpassing",
    "volleys",
    "marking",
    "standingtackle",
    "slidingtackle",
    "dribbling",
    "curve",
    "freekickaccuracy",
    "longpassing",
    "ballcontrol",
    "shotpower",
    "jumping",
    "stamina",
    "strength",
    "longshots",
    "acceleration",
    "sprintspeed",
    "agility",
    "reactions",
    "balance",
    "aggression",
    "composure",
    "interceptions",
    "positioning",
    "vision",
    "penalties"
}

local pgs = get_players_in_player_growth_system()

local int_current_date = game_db_manager:get_table_record_field_value(
    readPointer("pCareerCalendarTableCurrentRecord"), "career_calendar", "currdate"
)

local current_date = {
    day = 1,
    month = 7,
    year = 2021
}

if int_current_date > 20080101 then
    local s_currentdate = tostring(int_current_date)
    current_date = {
        day = tonumber(string.sub(s_currentdate, 7, 8)),
        month = tonumber(string.sub(s_currentdate, 5, 6)),
        year = tonumber(string.sub(s_currentdate, 1, 4)),
    }
end

local first_record = game_db_manager.tables["players"]["first_record"]
local record_size = game_db_manager.tables["players"]["record_size"]
local written_records = game_db_manager.tables["players"]["written_records"]

local row = 0
local current_addr = first_record
local last_byte = 0
local is_record_valid = true

while true do
    if row >= written_records then
        break
    end
    current_addr = first_record + (record_size*row)
    last_byte = readBytes(current_addr+record_size-1, 1, true)[1]
    is_record_valid = not (bAnd(last_byte, 128) > 0)
    if is_record_valid then
        local playerid = game_db_manager:get_table_record_field_value(current_addr, "players", "playerid")
        if playerid > 0 then
            -- Randomize position
            local new_primary_pos = nil
            if random_position then
                new_primary_pos = math.random(0, 27)
                if new_primary_pos == 1 then
                    new_primary_pos = 5     -- Replace SW with CB
                end
                game_db_manager:set_table_record_field_value(current_addr, "players", "preferredposition1", new_primary_pos)
            end
            if random_attributes then
                local primary_pos = new_primary_pos or game_db_manager:get_table_record_field_value(current_addr, "players", "preferredposition1")

                -- Randomize attributes
                local new_ovr = 0
                local ovr_formula = OVR_FORMULA_2[primary_pos]
                
                for _, fld in ipairs(attributes_fields) do
                    local new_attr_val = math.random(attribute_min, attribute_max)
                    if ovr_formula[fld] then
                        new_ovr = new_ovr + (new_attr_val * ovr_formula[fld])
                    end

                    game_db_manager:set_table_record_field_value(current_addr, "players", fld, new_attr_val)
                    playersEditorFormManager:update_xp_in_player_growth_system(playerid, pgs[playerid], fld, new_attr_val)
                end
                new_ovr = math.floor(new_ovr) + game_db_manager:get_table_record_field_value(current_addr, "players", "modifier")
                game_db_manager:set_table_record_field_value(current_addr, "players", "overallrating", new_ovr)
                local new_pot = new_ovr + 5
                if new_pot >= 99 then
                    new_pot = 99
                end
                game_db_manager:set_table_record_field_value(current_addr, "players", "potential", new_pot)
            end


            -- Randomize age
            if random_age then
                local playerbirthdate = game_db_manager:get_table_record_field_value(current_addr, "players", "birthdate")
                
                local bdate = days_to_date(playerbirthdate)
                
                local player_current_age = calculate_age(current_date, bdate)
                local new_age = player_current_age + math.random(age_change_range_min, age_change_range_max)

                if player_current_age ~= new_age then
                    if new_age > age_max then 
                        new_age = age_max
                    elseif new_age < age_min then
                        new_age = age_min
                    end

                    playerbirthdate = playerbirthdate + ((player_current_age - new_age) * 366)
                    game_db_manager:set_table_record_field_value(current_addr, "players", "birthdate", playerbirthdate)
                end
            end

            -- Randomize nationality
            if random_nationality then
                local new_nationality = nationalities[math.random(1, #nationalities-1)]
                game_db_manager:set_table_record_field_value(current_addr, "players", "nationality", new_nationality)
            end
        end
    end
    row = row + 1
end

showMessage("Done")